lib/Parse/ParseDecl.cpp

```diff
@@ -276,7 +277,8 @@ void Parser::parseTopLevelItems(SmallVectorImpl<ASTNode> &items) {
                !Context.Diags.hadAnyError() &&
                swift_ASTGen_emitParserDiagnostics(&Context.Diags,
                                                   SF.exportedSourceFile,
-                                                  /*emitOnlyErrors=*/true)) {
+                                                  /*emitOnlyErrors=*/true,
+                                                  /*downgradePlaceholderErrors=*/true)) {
```

Comment:
You should be getting whether to downgrade placeholder errors to warnings from the `LangOpts`. Something like the following

```suggestion
                                                  Context.LangOpts.WarnOnEditorPlaceholder)) {
```

same below